receipt_layoutlm/receipt_layoutlm/data_loader.py

```
@@ -330,12 +332,51 @@ class MergeInfo:
+        local_path = os.path.join(cache_dir, f"{image_id}_{receipt_id}.png")
+        if os.path.exists(local_path):
+            skipped += 1
+            continue
+        try:
+            receipt = dynamo.get_receipt(image_id, receipt_id)
+            if receipt and receipt.raw_s3_bucket and receipt.raw_s3_key:
+                s3.download_file(receipt.raw_s3_bucket, receipt.raw_s3_key, local_path)
+                downloaded += 1
+            else:
+                logger.warning("Receipt %s/%d has no S3 location, will use placeholder", image_id, receipt_id)
+        except Exception as e:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)